src/test/typescript-service-helpers.ts

```diff
@@ -0,0 +1,1525 @@
+					range: {
+						start: {
+							line: 24,
+							character: 14
+						},
+						end: {
+							line: 24,
+							character: 25
+						}
+					}
+				}]);
+			});
+		});
+	});
+
+	describe('getCompletions()', function () {
+		beforeEach(<any>initializeTypeScriptService(new TestFileSystem(new Map([
+			['file:///a.ts', [
+				'class A {',
+				'	/** foo doc*/',
+				'    foo() {}',
+				'	/** bar doc*/',
+				'    bar(): number { return 1; }',
+				'	/** baz doc*/',
+				'    baz(): string { return ""; }',
+				'	/** qux doc*/',
+				'    qux: number;',
+				'}',
+				'const a = new A();',
+				'a.'
+			].join('\n')],
+			['file:///uses-import.ts', [
+				'import * as i from "./import"',
+				'i.'
+			].join('\n')],
+			['file:///import.ts', '/** d doc*/ export function d() {}'],
+			['file:///uses-reference.ts', [
+				'/// <reference path="reference.ts" />',
+				'let z : foo.'
+			].join('\n')],
+			['file:///reference.ts', [
+				'namespace foo {',
+				'	/** bar doc*/',
+				'	export interface bar {}',
+				'}'
+			].join('\n')],
+			['file:///empty.ts', '']
+		]))));
+		it('produces completions in the same file', <any>async function (this: TestContext) {
+			const result = await this.service.getCompletions({
+				textDocument: {
+					uri: 'file:///a.ts'
+				},
+				position: {
+					line: 11,
+					character: 2
+				}
+			});
+			assert.equal(result.isIncomplete, false);
+			assert.sameDeepMembers(result.items, [
+				{
+					label: 'bar',
+					kind: CompletionItemKind.Method,
+					documentation: 'bar doc',
+					sortText: '0',
+					detail: '(method) A.bar(): number'
+				},
+				{
+					label: 'baz',
+					kind: CompletionItemKind.Method,
+					documentation: 'baz doc',
+					sortText: '0',
+					detail: '(method) A.baz(): string'
+				},
+				{
+					label: 'foo',
+					kind: CompletionItemKind.Method,
+					documentation: 'foo doc',
+					sortText: '0',
+					detail: '(method) A.foo(): void'
+				},
+				{
+					label: 'qux',
+					kind: CompletionItemKind.Property,
+					documentation: 'qux doc',
+					sortText: '0',
+					detail: '(property) A.qux: number'
+				}
+			]);
+		});
+		it('produces completions for imported symbols', <any>async function (this: TestContext) {
+			const result = await this.service.getCompletions({
+				textDocument: {
+					uri: 'file:///uses-import.ts'
+				},
+				position: {
+					line: 1,
+					character: 2
+				}
+			});
+			assert.deepEqual(result, {
+				isIncomplete: false,
+				items: [{
+					label: 'd',
+					kind: CompletionItemKind.Function,
+					documentation: 'd doc',
+					detail: 'function d(): void',
+					sortText: '0'
+				}]
+			});
+		});
+		it('produces completions for referenced symbols', <any>async function (this: TestContext) {
+			const result = await this.service.getCompletions({
+				textDocument: {
+					uri: 'file:///uses-reference.ts'
+				},
+				position: {
+					line: 1,
+					character: 13
```

Comment:
It's the same test file adapted, didn't change the expected values (except for legitimate cases where eg `range` was missing or the return value of `getDefinition` has to be an array)